cloudflare-ddns

```diff
@@ -138,12 +148,24 @@ def update_dns(subdomain, auth, ipv4_address, ipv6_address):
     """
     # Extract the domain
     domain = get_fld(subdomain, fix_protocol=True)
+    
     # Find the zone ID corresponding to the domain
-    zone_resp = requests.get(CLOUDFLARE_ZONE_QUERY_API, headers=auth, timeout=6)
-    if zone_resp.status_code != 200:
-        print('Authentication error: make sure your email and API key are correct. To set new values, run cloudflare-ddns --configure')
-        return
-    zone_names_to_ids = {zone['name']: zone['id'] for zone in zone_resp.json()['result']}
+    cur_page = 1
+    zone_names_to_ids = {}
+    while True:
```

Comment:
Any server-side rate limiting that might need to be considered here?

Reply:
The API limit is 1,200 requests in 5 minutes so I wouldn't be worried about hitting that limit while going through the pages. A nice feature for the future would be to run this function once towards the end of the configuration process and cache the `zone_names_to_ids` array in the config file, for example. This would help speed up updates for accounts that have a large number of domains and pretty much guarantee that the API rate limit isn't hit when doing updates. If there's a significant number of users with more than 60,000 domains (1200 pages * 50 per page) then I'd start looking into rate limiting the requests.

Reply:
Nice, should be fine as-is. Sorry, I wasn't suggesting you change this behavior, just wanted to make sure that it will work as expected for reasonable use cases/customers.